kombu/utils/limits.py

```diff
@@ -21,32 +21,21 @@ class TokenBucket:
         section of any multithreaded code.
     """
 
-    #: The rate in tokens/second that the bucket will be refilled.
-    fill_rate = None
+    def __init__(
+        self,
+        fill_rate: Union[str, int, float],
+        capacity: int = 1
+    ) -> None:
+        #: Maximum number of tokens in the bucket.
+        self.capacity = self._tokens = float(capacity)
```

Comment:
token capacity is discrete and logically is an int, but cast it to float here to make `min` and `max` calls simpler further down on the class.